tests/js/sentry-test/reactTestingLibrary.tsx

```diff
@@ -116,38 +127,74 @@ function makeAllTheProviders(options: ProviderOptions) {
       getCurrentLocation: jest.fn(() => ({pathname: '', query: {}})),
     });
 
-    // By default react-router 6 catches exceptions and displays the stack
-    // trace. For tests we want them to bubble out
-    function ErrorBoundary(): React.ReactNode {
-      throw useRouteError();
-    }
-
-    const routes: RouteObject[] = [
-      {
-        path: '*',
-        element: wrappedContent,
-        errorElement: <ErrorBoundary />,
-      },
-    ];
-
-    const memoryRouter = createRouter({
-      future: {v7_prependBasename: true},
-      history,
-      routes,
-    }).initialize();
-
     return (
       <CacheProvider value={{...cache, compat: true}}>
         <ThemeProvider theme={lightTheme}>
           <QueryClientProvider client={makeTestQueryClient()}>
-            <RouterProvider router={memoryRouter} />
+            {wrappedContent}
           </QueryClientProvider>
         </ThemeProvider>
       </CacheProvider>
     );
   };
 }
 
+function makeRouter({
+  children,
+  history,
+}: {
+  history: MemoryHistory;
+  children?: React.ReactNode;
+}) {
+  // By default react-router 6 catches exceptions and displays the stack
+  // trace. For tests we want them to bubble out
+  function ErrorBoundary(): React.ReactNode {
+    throw useRouteError();
+  }
+
+  const routes: RouteObject[] = [
+    {
+      path: '*',
+      element: children,
+      errorElement: <ErrorBoundary />,
+    },
+  ];
+
+  const router = createRouter({
+    future: {v7_prependBasename: true},
+    history,
+    routes,
+  }).initialize();
+
+  return router;
+}
+
+class TestRouter {
+  private router: Router;
+
+  constructor(router: Router) {
+    this.router = router;
+  }
+
+  get location() {
+    return this.router.state.location;
+  }
+
+  navigate = (to: To | number, opts?: RouterNavigateOptions) => {
+    rtl.act(() => {
+      if (typeof to === 'number') {
+        this.router.navigate(to);
+      } else {
+        this.router.navigate(to, opts);
+      }
+    });
+  };
+}
```

Comment:
This is new-style navigate right? Not to be confused with our `useNavigate` shim that is an abomination of react-router 3 style `LocationDescriptor` and new style options

Reply:
Yup this is a v6 navigate